The data science department of the company is asking us for help with some text for training on a Deep learning model.
We need to:
- Isolate the words
- Calculate the total number of words
- Calculate the average of words repetition
- Count of times the word appears in alphabetical order
- count of times the most frequent and least frequent word appears.

In [1]:
#Install and import the library "lorem"
!pip install lorem

In [2]:
import lorem

In [3]:
#Generate a random text:
lorem.text()

'Velit dolore ipsum modi labore adipisci ipsum neque. Labore labore velit velit ut dolor numquam. Tempora consectetur aliquam ut porro tempora ut. Ut tempora dolor sit dolore quisquam amet quiquia. Labore adipisci quisquam etincidunt etincidunt est amet adipisci. Labore adipisci aliquam eius. Dolore sit numquam quisquam porro aliquam voluptatem magnam.\n\nEst numquam voluptatem velit. Quaerat non quaerat modi. Neque tempora voluptatem dolore. Etincidunt dolor magnam magnam aliquam. Tempora porro dolore modi.\n\nDolorem voluptatem non quaerat eius neque. Ut eius sit ipsum porro ut dolore modi. Aliquam numquam etincidunt etincidunt dolorem est quaerat sit. Etincidunt tempora sit neque labore. Labore numquam dolore eius quisquam porro dolore magnam.'

In [15]:
#We need to save the text in a variable, if not is going to change every time we generate a text.
text = lorem.text()
text

'Modi quaerat quisquam labore non eius dolorem etincidunt. Aliquam ipsum magnam magnam. Labore sed consectetur velit. Voluptatem etincidunt velit dolor. Labore voluptatem dolorem voluptatem est est. Velit ipsum etincidunt magnam labore.\n\nDolor voluptatem sit quaerat velit quisquam. Etincidunt eius adipisci etincidunt ipsum dolor modi amet. Ipsum est velit neque sed quaerat. Quaerat numquam quaerat quiquia tempora quisquam. Velit est etincidunt tempora modi. Dolor modi ut non.\n\nAmet velit adipisci adipisci numquam. Dolore magnam est adipisci. Neque etincidunt neque consectetur. Voluptatem porro labore dolorem ut. Eius adipisci velit velit magnam magnam. Aliquam ut dolorem quiquia velit.\n\nAmet ut voluptatem tempora tempora est tempora consectetur. Voluptatem dolor voluptatem tempora quaerat aliquam non quaerat. Ipsum non etincidunt numquam dolor aliquam dolore. Quisquam ut ut quisquam sit sit modi sit. Non adipisci non dolor.\n\nNeque velit etincidunt velit. Sed eius velit sed eius

In [8]:
#Before counting the words, we need to remove dots for spaces, or any special characters, and after transform upper letters to lowers.
#For this we are going to use the need to use "replace" and "lower"

In [18]:
clean_text = text.replace ('.','')

In [21]:
clean_text = clean_text.lower()

In [23]:
clean_text

'modi quaerat quisquam labore non eius dolorem etincidunt aliquam ipsum magnam magnam labore sed consectetur velit voluptatem etincidunt velit dolor labore voluptatem dolorem voluptatem est est velit ipsum etincidunt magnam labore\n\ndolor voluptatem sit quaerat velit quisquam etincidunt eius adipisci etincidunt ipsum dolor modi amet ipsum est velit neque sed quaerat quaerat numquam quaerat quiquia tempora quisquam velit est etincidunt tempora modi dolor modi ut non\n\namet velit adipisci adipisci numquam dolore magnam est adipisci neque etincidunt neque consectetur voluptatem porro labore dolorem ut eius adipisci velit velit magnam magnam aliquam ut dolorem quiquia velit\n\namet ut voluptatem tempora tempora est tempora consectetur voluptatem dolor voluptatem tempora quaerat aliquam non quaerat ipsum non etincidunt numquam dolor aliquam dolore quisquam ut ut quisquam sit sit modi sit non adipisci non dolor\n\nneque velit etincidunt velit sed eius velit sed eius voluptatem dolor tempor

In [24]:
#The text is in STRING format, now we need to separate them to analyze. We need to use SPLIT
words = clean_text.split()
words

['modi',
 'quaerat',
 'quisquam',
 'labore',
 'non',
 'eius',
 'dolorem',
 'etincidunt',
 'aliquam',
 'ipsum',
 'magnam',
 'magnam',
 'labore',
 'sed',
 'consectetur',
 'velit',
 'voluptatem',
 'etincidunt',
 'velit',
 'dolor',
 'labore',
 'voluptatem',
 'dolorem',
 'voluptatem',
 'est',
 'est',
 'velit',
 'ipsum',
 'etincidunt',
 'magnam',
 'labore',
 'dolor',
 'voluptatem',
 'sit',
 'quaerat',
 'velit',
 'quisquam',
 'etincidunt',
 'eius',
 'adipisci',
 'etincidunt',
 'ipsum',
 'dolor',
 'modi',
 'amet',
 'ipsum',
 'est',
 'velit',
 'neque',
 'sed',
 'quaerat',
 'quaerat',
 'numquam',
 'quaerat',
 'quiquia',
 'tempora',
 'quisquam',
 'velit',
 'est',
 'etincidunt',
 'tempora',
 'modi',
 'dolor',
 'modi',
 'ut',
 'non',
 'amet',
 'velit',
 'adipisci',
 'adipisci',
 'numquam',
 'dolore',
 'magnam',
 'est',
 'adipisci',
 'neque',
 'etincidunt',
 'neque',
 'consectetur',
 'voluptatem',
 'porro',
 'labore',
 'dolorem',
 'ut',
 'eius',
 'adipisci',
 'velit',
 'velit',
 'magnam',
 'magnam',

In [25]:
type(words)

list

In [26]:
#How many words is the text have?
len(words)

159

In [27]:
#What is the average number of word repetitions?

In [28]:
#To count unique words we need to transform the words list into a set
unique_words = set(words)

In [29]:
unique_words

{'adipisci',
 'aliquam',
 'amet',
 'consectetur',
 'dolor',
 'dolore',
 'dolorem',
 'eius',
 'est',
 'etincidunt',
 'ipsum',
 'labore',
 'magnam',
 'modi',
 'neque',
 'non',
 'numquam',
 'porro',
 'quaerat',
 'quiquia',
 'quisquam',
 'sed',
 'sit',
 'tempora',
 'ut',
 'velit',
 'voluptatem'}

In [33]:
#we transform again the set into a list, because we can use more funtions
unique_words = list(unique_words)

In [34]:
#lenght of the new list
len(unique_words)

27

In [36]:
#Average of words repetition is the unique words divided by total of words
len(words) / len(unique_words)

5.888888888888889

In [38]:
round (len(words) / len(unique_words),2)

5.89

In [39]:
#If we want to check the repetition of certain words, we can do it:
print('sit', words.count('sit'))

sit 4


In [41]:
print('velit', words.count('velit'))
print('tempora', words.count('tempora'))
print('dolor', words.count('dolor'))
print('ipsum', words.count('ipsum'))

velit 13
tempora 7
dolor 8
ipsum 5


In [43]:
#Order the words alphabetically and count how many times it's repeating the words
type(unique_words)

list

In [48]:
unique_words.sort()
unique_words

['adipisci',
 'aliquam',
 'amet',
 'consectetur',
 'dolor',
 'dolore',
 'dolorem',
 'eius',
 'est',
 'etincidunt',
 'ipsum',
 'labore',
 'magnam',
 'modi',
 'neque',
 'non',
 'numquam',
 'porro',
 'quaerat',
 'quiquia',
 'quisquam',
 'sed',
 'sit',
 'tempora',
 'ut',
 'velit',
 'voluptatem']

In [51]:
#To show the words and how maby they repite, we need to create a dictionary
repetitions = {}

In [54]:
#Then, using the index function we can start completing the dictionary with every word and number of repeats.
repetitions [unique_words[0]]= words.count(unique_words[0])

In [55]:
repetitions

{'adipisci': 7}

In [58]:
repetitions [unique_words[1]]= words.count(unique_words[1])
repetitions [unique_words[2]]= words.count(unique_words[2])
repetitions [unique_words[3]]= words.count(unique_words[3])
repetitions [unique_words[4]]= words.count(unique_words[4])
repetitions [unique_words[5]]= words.count(unique_words[5])
repetitions [unique_words[6]]= words.count(unique_words[6])
repetitions [unique_words[7]]= words.count(unique_words[7])
repetitions [unique_words[8]]= words.count(unique_words[8])
repetitions [unique_words[9]]= words.count(unique_words[9])
repetitions [unique_words[10]]= words.count(unique_words[10])
repetitions [unique_words[11]]= words.count(unique_words[11])
repetitions [unique_words[12]]= words.count(unique_words[12])
repetitions [unique_words[13]]= words.count(unique_words[13])
repetitions [unique_words[14]]= words.count(unique_words[14])
repetitions [unique_words[15]]= words.count(unique_words[15])
repetitions [unique_words[16]]= words.count(unique_words[16])
repetitions [unique_words[17]]= words.count(unique_words[17])
repetitions [unique_words[18]]= words.count(unique_words[18])
repetitions [unique_words[19]]= words.count(unique_words[19])
repetitions [unique_words[20]]= words.count(unique_words[20])
repetitions [unique_words[21]]= words.count(unique_words[21])
repetitions [unique_words[22]]= words.count(unique_words[1])
repetitions [unique_words[23]]= words.count(unique_words[23])
repetitions [unique_words[24]]= words.count(unique_words[24])
repetitions [unique_words[25]]= words.count(unique_words[25])
repetitions [unique_words[26]]= words.count(unique_words[26])

In [59]:
repetitions

{'adipisci': 7,
 'aliquam': 4,
 'amet': 5,
 'consectetur': 3,
 'dolor': 8,
 'dolore': 4,
 'dolorem': 5,
 'eius': 5,
 'est': 7,
 'etincidunt': 10,
 'ipsum': 5,
 'labore': 6,
 'magnam': 6,
 'modi': 6,
 'neque': 5,
 'non': 6,
 'numquam': 4,
 'porro': 1,
 'quaerat': 7,
 'quiquia': 3,
 'quisquam': 5,
 'sed': 8,
 'sit': 4,
 'tempora': 7,
 'ut': 6,
 'velit': 13,
 'voluptatem': 9}

In [65]:
#How many times appear the most and the least frequent words?
repetitions.values()

dict_values([7, 4, 5, 3, 8, 4, 5, 5, 7, 10, 5, 6, 6, 6, 5, 6, 4, 1, 7, 3, 5, 8, 4, 7, 6, 13, 9])

In [62]:
min(repetitions.values())

1

In [63]:
max(repetitions.values())

13

In [66]:
sorted(repetitions.values())

[1,
 3,
 3,
 4,
 4,
 4,
 4,
 5,
 5,
 5,
 5,
 5,
 5,
 6,
 6,
 6,
 6,
 6,
 7,
 7,
 7,
 7,
 8,
 8,
 9,
 10,
 13]

Conclusion:
"Velit" was the most repetitive word in the text, and the least was "Porro", in 27 unique words. The average repetition was 5,89% from the total of 159 words.